File: Registroenvio/src/main/java/com/example/registroenvio/entity/Registroenvio.java
package com.example.registroenvio.entity;


import com.example.registroenvio.dto.ClienteDto;
import com.example.registroenvio.dto.VehiculoDto;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
public class Registroenvio  {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String titulo;
    private String descripcion;
    private LocalDateTime fecha;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "registro_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    public List<RegistroDetalle> registroDetalles;

    private Integer VehiculoId;
    @Transient
    private VehiculoDto vehiculoDto;

    private Integer ClienteId;
    @Transient
    private ClienteDto clienteDto;

    @Enumerated(EnumType.STRING)  // 0:pendiente 1:encamino 2:entregadoPara almacenar el nombre del estado como texto en la base de datos
    private EstadoEnvio estadoEnvio;
    public enum EstadoEnvio {
        PENDIENTE,
        EN_CAMINO,
        ENTREGADO
    }
}

File: app_vin/src/app/views/dashboard/setup/registroenvio/models/registroenvio.ts
import { Vehiculo } from '../../vehiculo/models/vehiculo';
import { Cliente } from '../../client/models/cliente';

export class RegistroEnvio {
    id!: number;
    titulo: string;
    descripcion: string;
    fecha: string; // Fecha en formato ISO
    vehiculoId: number; // Llave foránea para Vehículo
    clienteId: number; // Llave foránea para Cliente
    estadoEnvio: EstadoEnvio; // Enum para el estado del envío
    vehiculoDto?: Vehiculo; // Detalle del vehículo relacionado
    clienteDto?: Cliente; // Detalle del cliente relacionado
}

// Enumeración para el estado del envío
export enum EstadoEnvio {
    PENDIENTE = 'PENDIENTE',
    EN_CAMINO = 'EN_CAMINO',
    ENTREGADO = 'ENTREGADO',
}
